includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -517,6 +517,51 @@ public function save_payment_method_checkbox( $force_checked = false ) {
 		<?php
 	}
 
+	/**
+	 * Adds customer data to 'wcpay-checkout' when on 'Pay for Order' or 'Add Payment Method' pages.
+	 * Customer data is retrieved from order when on Pay for Order.
+	 * Customer data is retrieved from customer when on 'Add Payment Method'
+	 */
+	public function maybe_localize_customer_data() {
+		if ( ! isset( $_GET['pay_for_order'] ) && ! is_add_payment_method_page() ) {
+			return;
+		}
+
+		global $wp;
+		$user_email = '';
+		$firstname  = '';
+		$lastname   = '';
+
+		if ( isset( $_GET['pay_for_order'] ) && 'true' === $_GET['pay_for_order'] ) {
+			$order_id = wc_clean( $wp->query_vars['order-pay'] );
+			$order    = wc_get_order( $order_id );
+
+			if ( is_a( $order, 'WC_Order' ) ) {
+				$firstname  = $order->get_billing_first_name();
+				$lastname   = $order->get_billing_last_name();
+				$name       = $firstname . ' ' . $lastname;
```

Comment:
```suggestion
```
Unused var

Reply:
Thanks for finding the redundancy. I ended up removing $`firstname` and `$lastname` because the value that is used in the script is `$name`.